web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -448,6 +448,65 @@ const EncryptionSettingsForm = ({
   );
 };
 
+/**
+ * Allows to define snapshots enablement
+ * @component
+ *
+ * @param {object} props
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {onChangeFn} [props.onChange=noop] - On change callback
+ *
+ * @callback onChangeFn
+ * @param {object} settings
+ */
+const SnapshotsField = ({
+  settings,
+  onChange = noop
+}) => {
+  const rootVolume = settings.volumes.find((i) => i.mountPath === "/");
+
+  const initialChecked = rootVolume !== undefined && rootVolume.fsType === "Btrfs" && rootVolume.snapshots;
+  const [isChecked, setIsChecked] = useState(initialChecked);
+
+  if (rootVolume === undefined) { // no root volume is probably some error or still loading
+    return <Skeleton width="25%" />;
+  }
+
+  const switchState = (checked) => {
+    setIsChecked(checked);
+    onChange({ value: checked, settings });
+  };
+
+  const explanation = _("Uses btrfs for the root file system allowing to boot to a previous version of the system after configuration changes or software upgrades.");
+  const selector = (
+    <div>
+      <Switch
+        id="snapshots"
+        label={_("Use Btrfs Snapshots")}
+        isReversed
+        isChecked={isChecked}
+        onChange={switchState}
+      />
+      <div>
+        {explanation}
+      </div>
+    </div>
+  );
+  const enabledStatement = (
+    <div>
+      {_("Btrfs snapshots required by product.")}
+    </div>
+  );
```

Comment:
NP: I think it is better to use _internal components_ or to change the approach instead of this one. Don't get me wrong, it is ok in this case, but I believe that it is always _executing_ the code inside these parenthesis, no matter if finally you end up using _the component_ or not.

so, I would do something like

```js
const configurableSnapshots = rootVolume.outline.snapshotsConfigurable;
const forcedSnapshots = !configurableSnapshots && rootVolume.fsType === "Btrfs" && rootVolume.snapshots;

const SnapshotsToggle = () => (
    <>
      <Switch
        id="snapshots"
        label={_("Use Btrfs Snapshots")}
        isReversed
        isChecked={isChecked}
        onChange={switchState}
      />
      <div>
        {explanation}
      </div>
    </>
);

return (
  <div>
    <If condition={forcedSnapshots) then={_("Btrfs snapshots required by product.")} />
    <If condition={configurableSnapshots) then={<SnapshotsToggle />} />
  </div>
);
```